[WIP] Updating principles from Collab Cafe (#241)
sa-tre/satre-specification: docs/source/specification.md

:{ref}`Architectural Principles <satre_principles>`: The {term}`principles <architectural principle>` that all {ref}`TRE operator <infrastructure_roles>`s looking to use the specification should hold themselves accountable to.
:{ref}`Specification Pillars <satre_pillars>`: The broad areas of TRE provisioning the specification covers.

>                                                Each pillars is broken down into one or more {term}`capabilities <capability>`.

Reviewer: ```suggestion
                                               Each pillar is broken down into one or more {term}`capabilities <capability>`.
```

(satre_principles)=

### Architectural Principles

> Architectural principles influence and shape the way you design and deliver a SATRE-aligned TRE.

Reviewer 2: I'm not sure this is right, aren't these architectural principles more like what we had in mind when writing the spec, rather than what TRE developers/operators need to consider? That's why they were supposed to be brief originally.

Reviewer 2: @harisood I think we should decide what the purpose of this section is and who is supposed to read it - should a TRE operator looking to improve their TRE or a TRE developer starting from scratch read these principles before going to the spec? Some of the statements read like they should be spec statements, especially "Observability" - are we duplicating content here?

Reviewer: I think I agree- the "Implications" of these principles _are_ the statements in the rest of the spec, so I think we just need the statement and rationale here.

Does this need to be formal (heading, statement, rationale), or can we reformat this as:

```
#### Usability: A TRE instance that works for all users minimises barriers to use, and provides a productive and accessible analysis environment for research.

There is often a trade-off between increased operational security and the usability of a TRE.
In order to maintain productivity, a TRE must balance these two competing aims.
The design and configuration of a TRE should allow all individuals involved with a TRE to effectively fulfil their roles.
```

Author: > architectural principles more like what we had in mind when writing the spec, rather than what TRE developers/operators need to consider?

No these are for TRE people to consider. They're not hard and fast requirements.

> the "Implications" of these principles are the statements in the rest of the spec

Not always the case, for instance a lot of the usability ones are legit implcations. Agree for any should statements though - maybe these should be moved into the spec?

They are a set of guiding considerations that sit above any specific architectural requirement, and can be applied across the entire architecture.

Finally, the {ref}`TRE operator <infrastructure_roles>`s will need to consider different {ref}`roles <satre_roles>` with which individuals might interact with the TRE.

They consist of the following parts:
**Statement**: A singular sentence that summarises the principle
**Rationale**: Justification as to why this principle is important for the specification
**Implications**: Things you need to consider or do to practise this principle

#### Usability

##### Statement

> A TRE instance that works for everyone minimises barriers to use, and provides a productive and accessible analysis environment for research.

Author: Sure agreed!

> A TRE instance that works for everyone minimises barriers to use, and provides a productive and accessible analysis environment for research.

Author: ```suggestion
A TRE instance that works for all users minimises barriers to use, and provides a productive and accessible analysis environment for research.
```

Being transparent about the data held and the projects or organisations who access the data can also help maintain trust.

##### Implications

- Being as transparent as possible is key to building trust. TREs holding public data should practice transparency, for instance making adherence to any accreditation body, or framework (for instance the Five Safes) publicly available information, and making information on the projects or organisations which access their data available to impacted parties.
- A strong public engagement programme takes time and resource. TREs holding public data should consider allocating specific staffing to public engagement activities.
- TREs open to consultation by impacted parties should be auditable, and impacted parties should be part of any decision making processes. This may include the provision of documentation and educational resources for a diverse audience.
- Access to public sector data should be reviewed by an independent panel and follow agreed-upon governance to ensure projects using this data are in the public benefit, and provide clarity around any commercial access.

> #### Observability

Reviewer 2: Perhaps a more high level principle would be "Transparency" than "Observability" - I think logging user actions is just one possible method of increasing trust

Reviewer: I think this is about internal monitoring (both the technical aspects of the TRE, and the effectiveness of processes/IG). A consequence of this is increased trust, but I don't think that's the primary goal.
https://github.com/sa-tre/satre-specification/issues/173

##### Statement

Human initiated and automated processes resulting in change within the TRE should be observable.

##### Rationale

System/process observability is key to understanding whether your policies and controls are actually doing what is intended.

It also allows operators to continuously improve their systems and processes, measure their effectiveness, and identify the causes of incidents. Data can also be made available to other parties such as auditors, data subjects and data providers as part of the assurance process.

This applies to both technical systems and policies/processes.

##### Implications

> - In order to understand what is happening within the TRE, both automated and human initiated processes should generate sufficient data. Any generated data should follow standards for provenance and transparency for audit trails.

Reviewer: "generate sufficient data" sounds quite vague. Can we be more specific- e.g. Audit logs?

Author: ```suggestion
- In order to understand what is happening within the TRE, both automated and human initiated processes should generate sufficient data, for instance through audit logs. Any generated data should follow standards for provenance and transparency for audit trails.
```

- Different levels of observability may be needed for different users. Any data collected from an observability perspective should consider the needs of those who will use it, and minimise collection accordingly.
- There may be ethical and confidential issues to consider when implemeting the observability principle.

#### Standardisation

##### Statement

TREs should adhere to standards or well-known patterns wherever possible.

##### Rationale

Standardisation makes it easier to design, operate, use and understand TREs, and reduces duplication of work.
This includes making TREs easier to use, deploy, and audit.

TREs should be built in such a way that they do not restrict or prevent interoperability where this may be desirable in future, by identifying and avoiding or removing barriers to interoperability.

Standardisation is also linked to the public trust principle, as a standard approach to TRE provision will make it eaiser for impacted parties to understand how their data will be used within TREs.

##### Implications

> - Standards for TREs don't currently exist, and may be developed in the future. TRE teams should be prepared to adapt their provision to fit the latest agreed upon standard.

Reviewer: "agreed upon standard": Agreed upon by who? The UK TRE community? Stakeholders of that particular TRE? The SATRE governing board?

Author: No-one knows atm!

> - Standards for TREs don't currently exist, and may be developed in the future. TRE teams should be prepared to adapt their provision to fit the latest agreed upon standard.

Reviewer 2: ```suggestion
- Owing to the broad definition of {ref}`TREs <what_tre>`, there are no specific TRE technical standards. TRE {ref}`Developers and Operators <infrastructure_roles>` should be prepared identify the technical standards appropriate for them to work towards meeting whilst developing or maintaining their TREs.
```

> - Standards TREs adhere to may range in scope, including technical, operational and governance requirements. All agreed upon standards should align to ensure there is no contradiction in requirements.

Reviewer: ```suggestion
- Standards that TREs adhere to may range in scope, including technical, operational and governance requirements. All agreed upon standards should align to ensure there is no contradiction in requirements.
```

> - Standards that TREs adhere to may range in scope, including technical, operational and governance requirements. All agreed upon standards should align to ensure there is no contradiction in requirements.

Reviewer 2: ```suggestion
- Standards that TREs adhere to may range in scope, including technical, operational and governance requirements.
- TRE {ref}`Developers and Operators <infrastructure_roles>` should ensure that when they aim to meet multiple standards, those standards align with one another to ensure there is no contradiction in requirements. 
```

> - Any developments in standards should reuse or incorporate pre-existing content and resources wherever possible.

Reviewer 2: ```suggestion

```

I don't think it's clear what "developments in standards" means here

Author: Happy to remove
